draft-ietf-quic-transport.md

```diff
@@ -1451,29 +1454,276 @@ failure.  If integrity protection is performed by QUIC, QUIC MUST abort the
+validation, since the acknowledgment can be spoofed by a malicious client.
+
+For path validation to be successful, a PATH_RESPONSE frame MUST be received
+from the same remote address to which the corresponding PATH_CHALLENGE was
+sent. If a PATH_RESPONSE frame is received from a different remote address than
+the one to which the PATH_CHALLENGE was sent, path validation is considered to
+have failed, even if the data matches that sent in the PATH_CHALLENGE.
+
+Additionally, the PATH_RESPONSE frame MUST be received on the same local address
+from which the corresponding PATH_CHALLENGE was sent.  If a PATH_RESPONSE frame
+is received on a different local address than the one from which the
+PATH_CHALLENGE was sent, path validation is considered to have failed, even if
+the data matches that sent in the PATH_CHALLENGE.  Thus, the endpoint considers
+the path to be valid when a PATH_RESPONSE frame is received on the same path
+with the same payload as the PATH_CHALLENGE frame.
+
+
+### Abandonment
+
+An endpoint SHOULD abandon path validation after sending some number of
+PATH_CHALLENGE frames or after some time has passed.  When setting this timer,
+implementations are cautioned that the new path could have a longer round-trip
+time than the original.
+
+Note that the endpoint might receive packets containing other frames on the new
+path, but a PATH_RESPONSE frame with appropriate data is required for path
+validation to succeed.
+
+If path validation fails, the path is deemed unusable.  This does not
+necessarily imply a failure of the connection - endpoints can continue sending
+packets over other paths as appropriate.  If no paths are available, an endpoint
+can wait for a new path to become available or close the connection.
+
+A path validation might be abandoned for other reasons besides
+failure. Primarily, this happens if a connection migration to a new path is
+initiated while a path validation on the old path is in progress.
+
+
 ## Connection Migration {#migration}
 
-A QUIC connection can be identified using the Destination Connection ID included
-in packets.  QUIC's consistent connection ID allows connections to survive
-changes to the client's IP and/or port, such as those caused by client or server
-migrating to a new network.  Connection migration allows a client to retain any
-shared state with a connection when they move networks.  This includes state
-that can be hard to recover such as outstanding requests, which might otherwise
-be lost with no easy way to retry them.
+QUIC allows connections to survive changes to endpoint addresses (that is, IP
+address and/or port), such as those caused by a endpoint migrating to a new
+network.  This section describes the process by which an endpoint migrates to a
+new address.
+
+An endpoint MUST NOT initiate connection migration before the handshake is
+finished and the endpoint has 1-RTT keys.
+
+This document limits migration of connections to new client addresses.
```

Comment:
It seems like this is the only paragraph that's client specific now.  Is there a reason to keep this paragraph now?

Reply:
There's more that's needed for a server migration, even though the mechanisms we have so far can be used in both.  I have a separate branch with the immediate-server-migration case, and we can follow on with "full" server migration if time permits.

Reply:
What Mike said.